### PR TITLE
Bug 2069685: Fix UI crash when pinned resource model does not exist

### PR DESCRIPTION
**Pull request by jeff-phillips-18**
Don't set the `validPinnedResources` until the resources have been validated.

### CODE REVIEW

**Comment by rhamilto** [comment]
@jeff-phillips-18, rebasing on `master` to pick up the changes in https://github.com/openshift/console/pull/11235 should help this get through the tests.

**Comment by rhamilto** [comment]
> @jeff-phillips-18, rebasing on `master` to pick up the changes in #11235 should help this get through the tests.

It turns out that's not correct.  CI will apply your changes to latest `master` automatically.

**Comment by invincibleJai** [comment]
@jeff-phillips-18 I verified this, so UI doesn't crash anymore but I still see it in pinned resource with 404

in BZ 

```
Expected results:

UI shows but without the pinned resource that doesn't exist
```

is this correct?

image

**Comment by jeff-phillips-18** [comment]
@invincibleJai No, it shouldn't show on the nav bar at all. I'll take a look

**Comment by jeff-phillips-18** [comment]
@invincibleJai Wondering if what you are seeing is a dev only issue. If I switch from a cluster with the resource model to one with out (restart my dev server connected to the new cluster) it shows the pin the first time, but it disappears on the next refresh. When it is still there the `modelFor` is still returning the model even though it ought not exist.

**Comment by invincibleJai** [comment]
> @invincibleJai Wondering if what you are seeing is a dev only issue. If I switch from a cluster with the resource model to one with out (restart my dev server connected to the new cluster) it shows the pin the first time, but it disappears on the next refresh. When it is still there the `modelFor` is still returning the model even though it ought not exist.

yeah it was dev only issue, will verify again and update

**Comment by invincibleJai** [comment]
/lgtm
/approve

Verified it works as expected

File: frontend/public/components/nav/perspective-nav.tsx [hunk 1]
```diff
@@ -49,7 +49,7 @@ const PerspectiveNav: React.FC<{}> = () => {
         idx={idx}
         resourceRef={resource}
         onChange={setPinnedResources}
-        onDrag={setValidPinnedResources}
+        onDrag={setPinnedResources}
```
Comment (debsmita1) [inline frontend/public/components/nav/perspective-nav.tsx:52]: @jeff-phillips-18 The intent was not to update the user-settings till the dragged resource is dropped on the target area

Comment (debsmita1) [reply frontend/public/components/nav/perspective-nav.tsx:52]: PTAL https://github.com/openshift/console/pull/11266

**Pull request merged**